dropwizard-async-http-client/src/main/java/smartthings/dw/asynchttpclient/AsyncHttpClientModule.java

```diff
@@ -57,6 +65,8 @@ AsyncHttpClient asyncHttpClient(Set<RequestFilter> requestFilters) {
 		DefaultAsyncHttpClientConfig.Builder builder = new DefaultAsyncHttpClientConfig.Builder();
```

Comment:
The current approach tried to be composable but also keeps implementation details independent. In this case `ZipkinAsyncHttpClientModule` extends `AsyncHttpClientModule` and the only thing it does it augment the environment information so that all the configuration code is concentrated in `AsyncHttpClientModule`, i.e. the builder is all configured in one place. I thought this approach was beneficial at the cost of not using the `instrument` call and requiring instantiating the classes.

The alternative is to be able to provide an intermediate step in the base class to be able to customize the builder before the client is built. I am not opposed to this approach, but I found it to be not as clean.